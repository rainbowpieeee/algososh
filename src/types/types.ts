import { ElementStates } from "./element-states";

export interface ICircleElement {
  adding?: boolean;
  deleting?: boolean;
  withoutArrow?: boolean;
  tail?: string;
  head?: string;
  char?: string | null;
  extraCircle?: {
    char: string;
  };
  state: ElementStates;
}