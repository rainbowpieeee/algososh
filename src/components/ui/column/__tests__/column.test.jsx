import renderer from "react-test-renderer";

import { Column } from "../column";

import { ElementStates } from "../../../../types/element-states";

describe("checking column with snapshots.", () => {
  it("column should be rendered", () => {
    const tree = renderer.create(<Column />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("column in default state shoud be rendered", () => {
    const tree = renderer
      .create(<Column state={ElementStates.Default} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("column in changing state shoud be rendered", () => {
    const tree = renderer
      .create(<Column state={ElementStates.Modified} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("column in modified state shoud be rendered", () => {
    const tree = renderer
      .create(<Column state={ElementStates.Changing} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});