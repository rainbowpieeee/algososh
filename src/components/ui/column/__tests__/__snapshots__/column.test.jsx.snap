// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checking column with snapshots. column in changing state shoud be rendered 1`] = `
<div
  className="content "
>
  <div
    className="column modified"
    style={
      Object {
        "height": 1,
      }
    }
  />
  <p
    className="text text_type_column text_color_input mt-3"
  />
</div>
`;

exports[`checking column with snapshots. column in default state shoud be rendered 1`] = `
<div
  className="content "
>
  <div
    className="column default"
    style={
      Object {
        "height": 1,
      }
    }
  />
  <p
    className="text text_type_column text_color_input mt-3"
  />
</div>
`;

exports[`checking column with snapshots. column in modified state shoud be rendered 1`] = `
<div
  className="content "
>
  <div
    className="column changing"
    style={
      Object {
        "height": 1,
      }
    }
  />
  <p
    className="text text_type_column text_color_input mt-3"
  />
</div>
`;

exports[`checking column with snapshots. column should be rendered 1`] = `
<div
  className="content "
>
  <div
    className="column default"
    style={
      Object {
        "height": 1,
      }
    }
  />
  <p
    className="text text_type_column text_color_input mt-3"
  />
</div>
`;
